Auto-generate OSE inventory file

JayKayy opened the pull request:
#### What does this PR do?

This role is used to generate a valid inventory file for an Openshift Enterprise install
#### How should this be manually tested?

Simply use the inventory file in the `openshift_install/test` dir(change and add to it as you wish) and run it with the playbook. Then check the inventory file in the dir you ran it from for inventory_<env_id> for your ose installer inventory file.
[jkwiatko@fedora23 openshift_install]$ ansible-playbook -i test/ose-provision openshift_install.yaml 
#### Is there a relevant Issue open for this?

N/A
#### Who would you like to review this?

/cc @etsauer  @sabre1041

Commit:
testing complete of inventory generation

etsauer commented on rhc-ose-ansible/integration.yaml:
```diff
@@ -0,0 +1,23 @@
+---
+- hosts: localhost
+  vars_prompt:
```
@JayKayy i don't think there's any reason to prompt for these. Just put them in your input inventory file, and show that they get output to the generated inventory file.

JayKayy replied:
@etsauer So this is where i'm slightly confused as i thought that the point was to not have an "inventory file"?

etsauer replied:
@JayKayy the point is not to avoid having an inventory file. The point is that we will be feeding in an rhc-ose specific inventory file for our stuff... and as part of the provisioning process, we want to "produce" an inventory file that looks like what you'd expect for an openshift-ansible inventory. This is for the purposes of debugging/troubleshooting openshift-ansible stuff vs rhc-ose stuff.

Basically the rhc-ose inventory will have a bunch of openshift installer vars, but also a bunch of stuff that has nothing to do with the openshift installer. we want to separate those openshift installer vars out to a separate inventory.

etsauer commented on rhc-ose-ansible/roles/openshift_generate_inventory/meta/main.yml:
```diff
@@ -0,0 +1,6 @@
+#---
```
did you mean to leave these commented out?

etsauer commented:
@JayKayy i think I would rename this role to just 'openshift-install'

oybed commented on rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2:
```diff
@@ -0,0 +1,67 @@
+masters
+nodes
+etcd
+lb
```
This one should probably have the same conditional as below - i.e. "{% if openshift_lbs is defined %}"

oybed commented on rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2:
```diff
@@ -0,0 +1,67 @@
+[OSEv3:children]
+masters
+nodes
+etcd
```
This one should probably have the same conditional as below - i.e.: "{% if openshift_etcds is defined %}"

Commit:
removed role dependency and fixed template bugs

Commit:
Update ose-provision

JayKayy commented:
Updated based on your feedback.
Changes:
- Removed role dependencies based on advice from ansible slack channel as how ansible handles role dependency is likely to change.
- Fixed a few bugs in the template (pointed out by @oybed) causing etcd and lb groups to appear regardless of actual group presence.
- added appropriate variables to ose-provision inventory file
- Changed role name to `openshift_install` ( @etsauer )
- Moved playbook to playbooks dir and added access to role directory

etsauer commented on rhc-ose-ansible/inventory/ose-provision:
```diff
@@ -9,9 +8,15 @@ openstack_key_name=""
+#openshift_openstack_master_storage_size=10
+#openshift_openstack_node_storage_size=10
+#env_id=
+inv_cluster_method=native
```
I think we want a more obvious prefix for these, like 'openshift-ansible_'

JayKayy replied:
Sure agreed. I'm hesitant to go with `openshift_ansible...` as we don't want our vars to overload/conflict with variables they set or we may run into compatibility issues with some of their roles/playbooks

etsauer replied:
cc: @detiber, want to get your opinion on this. We're basically using this playbook & role as a "decoupler" from our playbooks and the openshift-ansible stuff.

detiber replied:
We won't use the openshift_ansible prefix for variables, but that also leads to very long names...  you could use rhc_ose_ as a prefix for a bit shorter name that will not conflict.

etsauer commented:
Do we like the naming convention of the playbooks in rhc-ose-ansible/playbooks/registry? If so I think we should stick with that. I.e.

```
./rhc-ose-ansible/playbooks
  - ./openshift
    - ./byo.yml <-- this would be @JayKayy's current playbook in this PR
    - ./provision-openstack.yml <-- this would take the place of what's currently at ./rhc-ose-ansible/ose-provision.yml
```

cc: @oybed @sabre1041

detiber commented:
If these playbooks will only be used against the OpenStack instances being created, I would suggest moving away from generating an inventory file and rather taking advantage of a dynamic inventory file and mapping groups based on instance tags to the byo groups.

An example of how to invoke the playbook and map variables is here: https://github.com/2015-Middleware-Keynote/demo-ansible/blob/master/playbooks/openshift_setup.yml#L56-L98

An example of the group mapping can be seen here: https://github.com/2015-Middleware-Keynote/demo-ansible/blob/master/playbooks/group_setup.yml

It can also be done through a static inventory file like this:

```
---
[OSEv3:children]
masters
nodes
etcd

[masters:children]
tag_openshift-demo-jdetiber-host-type_master

[etcd:children]
tag_openshift-demo-jdetiber-host-type_master

[nodes:children]
tag_openshift-demo-jdetiber-host-type_master
tag_openshift-demo-jdetiber-host-type_node


[tag_openshift-demo-jdetiber-host-type_master]

[tag_openshift-demo-jdetiber-host-type_node]
```

In this case, you would have both the dynamic inventory script and the static inventory file in the same directory and would use the common directory as the inventory.

Commit:
added inventory destination variable and test dir

Commit:
Merge branch 'integration' of github.com:JayKayy/rhc-ose into integration

Commit:
renamed to openshift-install and adjust htpasswd file to /etc/origin/openshift-passwd

Commit:
rhc-htpasswd role

Commit:
correcting dependency task

Commit:
fixed test inv

Commit:
applying fix to inventory file

Commit:
removing inventory file

Commit:
Merge branch 'integration' of github.com:JayKayy/rhc-ose into integration

Commit:
adding ose-provision

Commit:
cleaing out unnecessary files

Commit:
removing extras

Commit:
fixed merge conflict in inventory

Commit:
merged

JayKayy commented on rhc-ose-ansible/inventory/ose-provision:
```diff
@@ -2,13 +2,14 @@
 #openstack_flavor_name="m1.medium"
-#openstack_image_name="ose3_1-base"
+openstack_image_name="_OS1_rhel-guest-image-7.2-20151102.0.x86_64.qcow2"
 #openshift_openstack_master_storage_size: 10
```
these should probably be changed to 
`openshift_openstack_master_storage_size=10`

Commit:
fixing role name

etsauer commented:
Gonna close this PR, as it's no longer relevant.
